src/wp-includes/theme.php

```diff
@@ -4090,7 +4090,6 @@ function create_initial_theme_features() {
  *
  * @return boolean Whether the current theme is a block-based theme or not.
  */
-function wp_is_block_template_theme() {
-	return is_readable( get_theme_file_path( '/block-templates/index.html' ) ) ||
-		is_readable( get_theme_file_path( '/templates/index.html' ) );
+function wp_is_block_theme() {
```

Comment:
@anton-vlasenko As Gutenberg has to support WP 5.7 and 5.8, how will renaming this function impact the plugin? Are you planning to rename it in Gutenberg too and deprecate the original name? Or does renaming not affect Gutenberg?

Reply:
@hellofromtonya Companion issue is [here](https://github.com/WordPress/gutenberg/issues/37162) and [the PR for that issue](https://github.com/WordPress/gutenberg/pull/37161/files) seems to have been updated in line with this page.

Reply:
Thanks @costdev for pointing that out.

Reply:
@hellofromtonya It shouldn't impact the plugin. I think it's safe to deprecate the original name because Gutenberg doesn't call that function directly.
Yes, I'm planning to rename it in Gutenberg. I've created [a PR that renames](https://github.com/WordPress/gutenberg/pull/37161) it.
IMO it doesn't make sense to keep both `wp_is_block_theme` and `wp_is_block_template_theme` functions in WordPress Core.
I think we still have time to rename it before the Gutenberg 12.1.0 release. Even if it doesn't make it into the Gutenberg 12.1.0, Gutenberg should work just fine without it.

Reply:
Thanks @anton-vlasenko. Part of the consolidation discuss is to help with the backports from Gutenberg to Core. So wanted to make sure whatever happens here also happens in the Gutenberg repo too, i.e. code matches.

Reply:
Thank you for making sure, @hellofromtonya.
I 100% agree.
We need to be careful to make sure we don't break Gutenberg.